rotkehlchen/chain/ethereum/tokens.py

```diff
@@ -61,65 +59,82 @@
-                    address=address,
-                    tokens=chunk,
-                    balances=balances,
-                    token_usd_price=token_usd_price,
-                    call_order=(ETHERSCAN_NODE,),
-                )
-
-        # now that detection happened we also have to save it in the DB for the address
-        self.db.save_tokens_for_address(write_cursor, address, list(balances.keys()))
-
-        return balances
-
-    def query_tokens_for_addresses(
-            self,
-            addresses: List[ChecksumEthAddress],
-            force_detection: bool,
-    ) -> TokensReturn:
-        """Queries/detects token balances for a list of addresses
-
-        If an address's tokens were recently autodetected they are not detected again but the
-        balances are simply queried. Unless force_detection is True.
+        """Queries amounts of multiple tokens for an account
 
-        Returns the token balances of each address and the usd prices of the tokens
+        May raise:
+        - RemoteError if an external service such as Etherscan is queried and
+          there is a problem with its query.
+        - BadFunctionCallOutput if a local node is used and the contract for the
+          token has no code. That means the chain is not synced
         """
         log.debug(
-            'Querying/detecting token balances for all addresses',
-            force_detection=force_detection,
+            'Querying ethereum chain for multi token account balances',
+            eth_address=account,
+            tokens_num=len(tokens),
         )
-        with self.db.user_write() as cursor:
-            ignored_assets = self.db.get_ignored_assets(cursor)
+        result = ETH_SCAN.call(
+            ethereum=self.ethereum,
+            method_name='tokensBalance',
+            arguments=[account, [x.ethereum_address for x in tokens]],
+            call_order=call_order,
+        )
+        amounts: Dict[EthereumToken, FVal] = defaultdict(FVal)
+        for token_amount, token in zip(result, tokens):
+            if token_amount == 0:
+                continue
 
+            normalized_amount = token_normalized_value(token_amount, token)
+            log.debug(
+                f'Found {token.symbol}({token.ethereum_address}) token balance for '
+                f'{account} and amount {normalized_amount}',
+            )
+            amounts[token] += normalized_amount
+        return amounts
+
+    def _query_chunks(
+        self,
+        account: ChecksumEthAddress,
+        chunks: Iterable[List[EthereumToken]],
+        querying_etherscan: bool,
+    ) -> Dict[EthereumToken, FVal]:
+        total_token_amounts: Dict[EthereumToken, FVal] = defaultdict(FVal)
+        for chunk in chunks:
+            if querying_etherscan is True:
+                call_order = [ETHERSCAN_NODE]
+            else:
+                call_order = self.ethereum.default_call_order()
+            new_token_amounts = self._get_token_amounts(
```

Comment:
adjust amount to balances here and below

Reply:
Done